src/Nethermind/Nethermind.Runner/scripts/app.ts

```diff
@@ -0,0 +1,400 @@
+const gasLimit = document.getElementById('gasLimit') as HTMLElement;
+const gasLimitDelta = document.getElementById('gasLimitDelta') as HTMLElement;
+
+const blockExtraData = document.getElementById('blockExtraData') as HTMLElement;
+const blockGasUsed = document.getElementById('blockGasUsed') as HTMLElement;
+const blockGasLimit = document.getElementById('blockGasLimit') as HTMLElement;
+const blockBlockSize = document.getElementById('blockBlockSize') as HTMLElement;
+const blockTimestamp = document.getElementById('blockTimestamp') as HTMLElement;
+const blockTransactions = document.getElementById('blockTransactions') as HTMLElement;
+const blockBlobs = document.getElementById('blockBlobs') as HTMLElement;
+
+//const ridgeDataLength = createRidgelinePlot(
+//  d3.select('#ridgeline-dataLength'),
+//  (d) => Math.log10( d.dataLength )// numeric field in the TxReceipt
+//);
+
+//const ridgeValue = createRidgelinePlot(
+//  d3.select('#ridgeline-value'),
+//  (d) => Math.log10(parseInt(d.value, 16) / 1000000000000000000) // parse from hex
+//);
+
+//const ridgeGasUsed = createRidgelinePlot(
+//  d3.select('#ridgeline-gasUsed'),
+//  (d) => Math.log10(parseInt(d.gasUsed, 16))
+//);
+
+//const ridgeEffectiveGasPrice = createRidgelinePlot(
+//  d3.select('#ridgeline-effectiveGasPrice'),
+//  (d) => Math.log10(parseInt(d.effectiveGasPrice, 16) / 1000000000)
+//);
+
+// Create a rolling box plot instance for effectiveGasPrice
+const boxPlotEGP = createRollingBoxPlot(
+  document.getElementById('boxplot-effectiveGasPrice'),
+  (d) => parseInt(d.effectiveGasPrice, 16) / 1_000_000_000, // convert to gwei
+  36 // Keep up to 36 blocks
+);
+
+const ansiConvert = new Convert();
+
+// We reuse these arrays for the sparkline. The length = 60 means we store 60 historical points.
+let seriesHashes: Datum[] = [];
+let seriesReceived: Datum[] = [];
+let seriesTxPool: Datum[] = [];
+let seriesBlock: Datum[] = [];
+let seriesDuplicate: Datum[] = [];
+
+let seriesTotalCpu: Datum[] = [];
+let seriesMemory: Datum[] = [];
+
+// Keep track of last values so we can compute TPS
+let lastReceived = 0;
+let lastTxPool = 0;
+let lastBlock = 0;
+let lastDuplicate = 0;
+let lastHashesReceived = 0;
+let lastNow = 0;
+
+function addCapped(array: Datum[], datum: Datum) {
+  array.push(datum);
+  if (array.length > 60) {
+    array.shift();
+  }
+}
+
+function updateText(element: HTMLElement, value: string): void {
+  if (element.innerText !== value) {
+    // Don't update the DOM if the value is the same
+    element.innerText = value;
+  }
+}
+
+// Initialize the Sankey flow
+const txPoolFlow = new TxPoolFlow('#txPoolFlow');
+
+// Number format
+const format = d3.format(',.0f');
+const formatDec = d3.format(',.1f');
+
+let txPoolNodes: INode[] = null;
+/**
+ * Main function to start polling data and updating the UI.
+ */
+function updateTxPool(txPool: TxPool) {
+
+  if (txPool.pooledTx == 0) {
+    document.getElementById("txPoolFlow").classList.add("not-active");
```

Comment:
If this element is going to always be present in the DOM is better to have a reference to it at the start instead of querying in a loop.